### DISCUSSION TITLE
Cannot scp from SolarWinds SCP server

### DISCUSSION THREAD

**Discussion by sburnwal** [libssh2/libssh2#1273] [Q&A]
**Describe the bug**
Trying to do scp a file from SolarWinds server but it fails. I have tried libssh2 1.11 and 1.9, sme behavior.

**To Reproduce**
Try to download a file from a SolarWinds server. 

A recipe or example code that reproduces the problem? A stack trace from a crash

**Expected behavior**
File scp should succeed.

**Version (please complete the following information):**
 - OS and version: Ubuntu 20.04 
 - libssh2 version: 1.11
 - crypto backend and version: OpenSSL 1.1.1w

**Additional context**
This is the sshd logs on SolarWinds. I do not know who is closing the connection (SolarWinds or libssh2). SCP of the file works fine with 'scp' cmd though.

```
OnReceive::Before ExchangeIdentification
ExchangeIdentification::Remote version is SSH-2.0-libssh2_1.9.0

OnReceive::After ExchangeIdentification
OnReceive::err = 0
OnReceive::packet type = 20
OnReceive::KexInit start
ChooseCryptos::Selected CScipher aes128-ctr
ChooseCryptos::Selected SCcipher aes128-ctr
ChooseCryptos::Selected CSmac hmac-sha2-256
ChooseCryptos::Selected SCmac hmac-sha2-256
OnReceive::KexInit end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 30
OnReceive::KexDHInit start
OnReceive::KexDHInit end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 21
OnReceive::NewKeys start
OnReceive::NewKeys end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 5
OnReceive::ServiceRequest start
ServiceRequest::read string
ServiceRequest::string is ssh-userauth
ServiceRequest::freed string
OnReceive::ServiceRequest end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 50
OnReceive::UserAuthRequest start
OnReceive::UserAuthRequest end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 50
OnReceive::UserAuthRequest start
OnReceive::UserAuthRequest end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 90
OnReceive::ChannelOpen start
ChannelOpen::service request = session
OnReceive::ChannelOpen end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 98
OnReceive::ChannelRequest start
ChannelRequest::request for exec
SCP command: scp -pf '/aci-n9000-dk9.15.2.7g.bin'
SSH2Channel::dtor
OnReceive::ChannelRequest end
OnReceive::packet type = 0
OnReceive::err = 0
OnReceive::packet type = 94
OnReceive::ChannelData start
OnReceive::ChannelData end
OnReceive::packet type = 0
OnReceive::Firing Receive Event for channel 0
OnClose::err = 0
```

**Comment by dfandrich** [comment]
>   • libssh2 version: 1.11
> 
> This is the sshd logs on SolarWinds
> 
> OnReceive::Before ExchangeIdentification
> ExchangeIdentification::Remote version is SSH-2.0-libssh2_1.9.0

These two versions don't match. Are you sure you're actually using 1.11 at
run-time?

Note that scp has been deprecated and is turned off by default in recent SSH
servers. Are you sure it's enabled in yours? The scp command-line tool now does
sftp under the hood so if that works it's not a definitive answer.

**Comment by sburnwal** [comment]
@dfandrich Thank you so much for replying back! I have the same output from libssh2 1.11 as well, though I sent the output for libssh2 1.9 at the time of raising this issue. I do see the lines below with ver 1.11.

```
OnReceive::Before ExchangeIdentification
ExchangeIdentification::Remote version is SSH-2.0-libssh2_1.11.0
```

I have follow ups:
(1) Anything you find suspicious in the above logs?

(2) We just set the protocol like below in the curl API. 
```
curl_easy_setopt(lCurl, CURLOPT_PROTOCOLS, CURLPROTO_SCP)
```
Are you suggesting I should switch to curl_easy_setopt(lCurl, CURLOPT_PROTOCOLS, CURLPROTO_SFTP) instead?

**Reply by vszakats** [reply]
It may be worth a try with the latest libssh2 1.11.1, to see if it makes a difference.

**Comment by sburnwal** [comment]
on the client (curl) side, I see these output, do not know why connection is getting dropped.
```
*   Trying 12.59.118.62:22...
* TCP_NODELAY set
* Connected to 12.59.118.62 (12.59.118.62) port 22 (#0)
* SSH MD5 fingerprint: 80ad54b6bc39c5f1f681cf46afe699b9
* SSH authentication methods available: password
* Initialized password authentication
* Authentication complete
* SSH CONNECT phase done
* Unexpected channel close
```

**Reply by Tasuka** [reply]
I can conform that Solarwind SFTP/SCP and Rebex tiny SFTP server on Windows 10 has SCP problem, SFTP is work verywell. Does any solution ?

**Comment by rmsh1216** [comment]
I suggest copying files from other SCP servers, such as sshd, to check whether the problem recurs. If the problem recurs, the detailed debug logs of sshd may explain the cause or provide more information to locate the cause.
